frontend/public/components/utils/webhooks.tsx

```diff
@@ -56,17 +56,30 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
   const [webhookSecrets, setWebhookSecrets] = React.useState<K8sResourceKind[]>([]);
+  const [webhookTriggers, setWebhookTriggers] = React.useState<WebhookTrigger[]>([]);
+  const [secretNames, setSecretNames] = React.useState<string[]>([]);
   const [secretErrors, setSecretErrors] = React.useState<string[]>([]);
   const [isLoaded, setLoaded] = React.useState(false);
 
-  const webhooks: WebhookTrigger[] = _.filter(triggers, ({ type }) => webhookTriggers.has(type));
-  const secretNames: string[] = _.uniq(
-    webhooks.reduce((acc: string[], webhook: WebhookTrigger): string[] => {
-      const triggerProperty = getTriggerProperty(webhook);
-      const secretName = _.get(webhook, [triggerProperty, 'secretReference', 'name']);
-      return secretName ? [...acc, secretName] : acc;
-    }, []),
-  );
+  React.useEffect(() => {
```

Comment:
I solved this in my eslint PR since the exhaustive deps rule was failing. I think it might be a good idea to separate the `webhooks` and `secretNames` updates into their own effects. That way the deps for webhooks can be set to `[triggers]` and then webhooks can be used as a dep for the secretNames effect. Something like this:

```tsx
  const [webhooks, setWebhooks] = React.useState<WebhookTrigger[]>([]);
  React.useEffect(() => {
    setWebhooks(_.filter(triggers, ({ type }) => webhookTriggers.has(type)));
  }, [triggers]);

  const [secretNames, setSecretNames] = React.useState<string[]>([]);
  React.useEffect(() => {
    setSecretNames(_.uniq(
      webhooks.reduce((acc: string[], webhook: WebhookTrigger): string[] => {
        const triggerProperty = getTriggerProperty(webhook);
        const secretName = _.get(webhook, [triggerProperty, 'secretReference', 'name']);
        return secretName ? [...acc, secretName] : acc;
      }, []))
    );
  }, [webhooks]);

```